pytrip/tripexecuter2/execute.py

```diff
@@ -0,0 +1,579 @@
+#
+#    Copyright (C) 2010-2017 PyTRiP98 Developers.
+#
+#    This file is part of PyTRiP98.
+#
+#    PyTRiP98 is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    PyTRiP98 is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with PyTRiP98.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+TODO: documentation here.
+"""
+import os
+import shutil
+import tarfile
+import copy
+
+import numpy as np
+from subprocess import Popen, PIPE
+
+try:
+    import paramiko
+except:
+    pass
+
+from pytrip.dos import DosCube
+from pytrip.let import LETCube
+from pytrip.vdx import VdxCube
+##import pytriplib
+import uuid
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class Execute(object):
+    def __init__(self, ctx):
+        """ Initialize the Execute class. 
+        :params CtxCube() ctx: the CT images as a regular pytrip.CtxCube() object. 
+        """
+
+        logger.debug("Initializing TripExecuter()")
+        self.__uuid__ = uuid.uuid4()  # for uniquely identifying this execute object
+        self.ctx = ctx
+        ##self.rbe = rbe
+        self.listeners = []
+        self.trip_bin_path = "TRiP98"  # where TRiP98 is installed, if not accessible in /usr/local/bin or similar
+        self.logfile_stdout = "trip98.stdout"
+        self.logfile_stderr = "trip98.stderr"
+        self.rsakey_local_path = "~/.ssh/id_rsa"
+        self._runtrip = True  # set this to False for a dry run without TRiP98 (for testing purposes)
+        ##self.run_dir = "" 
+        self.remote_dir = "/tmp/."  # remote directory where all will be run
```

Comment:
This is not good. On some machines user may not have access to `/tmp`, but only to some subdirectory.
Safer to use  sth like `mkdtemp`